player: add compatibility to players requiring TrackId for SetPosition

M4he opened the pull request:
Not sending the actual valid TrackId for the SetPosition D-Bus call is considered a violation of the MediaPlayer2.Player specification [1]. Some players might be fine without it but others require it. Clementine for instance, will ignore the SetPosition call when the trackId sent does not match the actual value.

The specification [1] says:

> TrackId — o ([Track_Id](https://specifications.freedesktop.org/mpris-spec/latest/Player_Interface.html#Simple-Type:Track_Id))
>
>The currently playing track's identifier.
>
>If this does not match the id of the currently-playing track, the call is ignored as "stale".
>
>/org/mpris/MediaPlayer2/TrackList/NoTrack is *not* a valid value for this argument.

[1] https://specifications.freedesktop.org/mpris-spec/latest/Player_Interface.html#Method:SetPosition

Functionality tested with Audacious and Clementine.

Commit:
player: add compatibility to players requiring TrackId for SetPosition

Not sending the actual valid TrackId for the SetPosition D-Bus call
is considered a violation of the MediaPlayer2.Player specification [1].
Some players might be fine without it but others require it.

[1] https://specifications.freedesktop.org/mpris-spec/latest/Player_Interface.html#Method:SetPosition

worron commented:
Thank you, this is quite right topic.

You are right in general, it always was half working solution. The problem is that according specification TrackId is `o` type, so awesomewm dbus module just strip it during parsing. We can't catch track id while listening mpris Properties if audio player really stricly follows specification. Funny coincidence, Audacious just doesn't care about track id on SetPosition, and Clemetine sends "mpris:trackid" as plain string by dbus.

So I added some workaround on top of your changes to guarantee track id info in any situation. Implemented double call doesn't look so good, but it's all I can find so far.

Thanks again for contribution.

M4he commented:
> [...] We can't catch track id while listening mpris Properties if audio player really stricly follows specification. Funny coincidence, Audacious just doesn't care about track id on SetPosition, and Clemetine sends "mpris:trackid" as plain string by dbus.
> 

Interesting, good catch!

> So I added some workaround on top of your changes to guarantee track id info in any situation. Implemented double call doesn't look so good, but it's all I can find so far.
> 

Thank you very much for revising my patch accordingly!
